src/main/scala/codacy/plugins/test/MultipleTests.scala

```diff
@@ -0,0 +1,143 @@
+            else CodacyCfg(patterns, Some(srcDir.pathAsString), extraValues)
+          } else FileCfg(Some(srcDir.pathAsString), None)
+        }
+        tools.exists { tool =>
+          val res = runTool(tool, srcDir, configuration)
+          res match {
+            case Failure(e) =>
+              Printer.red("Got failure in the analysis:")
+              e.printStackTrace()
+              false
+            case Success(results) =>
+              if (results.sameElements(expectedResults)) {
+                Printer.green(s"Got ${results.size} results.")
+                true
+              } else {
+                Printer.red("Tool results don't match expected results:")
+                Printer.red("Extra: ")
+                pprint.pprintln(results.diff(expectedResults), height = Int.MaxValue)
+                Printer.red("Missing:")
+                pprint.pprintln(expectedResults.diff(results), height = Int.MaxValue)
+                false
+              }
+          }
+        }
+      }
+    }
+  }
+
+  private def parseResultsXml(file: File) = {
+    for {
+      fileTag <- XML.loadFile(file.toJava) \\ "checkstyle" \\ "file"
+      fileName = fileTag \@ "name"
+      errorsTag <- fileTag \\ "error"
+      line = errorsTag \@ "line"
+      patternId = errorsTag \@ "source"
+      message = errorsTag \@ "message"
+      severity = errorsTag \@ "severity"
+      level = severity match {
+        case "info" => Level.Info
+        case "warning" => Level.Warn
+        case "error" => Level.Err
+        case _ => throw new Exception(s"$severity is not a valid level")
+      }
+    } yield PluginResult(patternId, fileName, line.toInt, message, level)
+  }
+
+  private def parsePatternsXml(file: File): (Set[Pattern], Option[Map[String, play.api.libs.json.JsValue]]) = {
+    val rootModule = XML.loadFile(file.toJava)
+    val extraValues = (rootModule \ "property").map { node =>
+      val v = node \@ "value"
+      val value = try {
+        Json.parse(v)
+      } catch {
+        case _: JsonParseException => // support non quoted strings
+          Json.parse(s""""$v"""")
+      }
+      (node \@ "name", value)
+    }.toMap
+    val patternsList = for {
+      patternTags <- rootModule \ "module"
+      patternId: String = patternTags \@ "name"
+      parameters = (patternTags \ "property").map { node =>
+        Parameter(node \@ "name", node \@ "value")
+      }.toSet
+    } yield Pattern(patternId, parameters)
+    (patternsList.toSet, if (extraValues.isEmpty) None else Some(extraValues))
+  }
+
+  private def runTool(tool: Tool, testDir: File, configuration: Configuration): Try[Set[PluginResult]] = {
+    val builder = Set.newBuilder[PluginResult]
+    for {
+      set <- tool.run(testDir, testDir.list.filter(_.isRegularFile).map(_.path).toSet, configuration)
+      toolResult <- set
+    } toolResult match {
+      case Issue(patternId, filename, message, level, category, location) =>
+        val line = location match {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Default case not specified.](https://app.codacy.com/app/Codacy/codacy-plugins-test/pullRequest?prid=4500290)